reco_utils/dataset/pandas_df_utils.py

```diff
@@ -117,55 +117,120 @@ def libffm_converter(df, col_rating=DEFAULT_RATING_COL, filepath=None):
         4       1  1:3:1  2:2:7  3:3:5.0  4:8:1
 
     Args:
-        df (pd.DataFrame): input Pandas dataframe.
-        col_rating (str): rating of the data.
         filepath (str): path to save the converted data.
 
-    Return:
-        pd.DataFrame: data in libffm format.
+    Attributes:
+        field_count (int): count of field in the libffm format data
+        feature_count (int): count of feature in the libffm format data
+        filepath (str or None): file path where the output is stored - it can be None or a string
     """
-    df_new = df.copy()
-
-    # Check column types.
-    types = df_new.dtypes
-    if not all([x == object or np.issubdtype(x, np.integer) or x == np.float for x in types]):
-        raise TypeError("Input columns should be only object and/or numeric types.")
-
-    field_names = list(df_new.drop(col_rating, axis=1).columns)
-
-    # Encode field-feature.
-    idx = 1
-    field_feature_dict = {}
-    for field in field_names:
-        if df_new[field].dtype == object:
-            for feature in df_new[field].values:
-                # Check whether (field, feature) tuple exists in the dict or not.
-                # If not, put them into the key-values of the dict and count the index.
-                if (field, feature) not in field_feature_dict:
-                    field_feature_dict[(field, feature)] = idx
-                    idx += 1
-
-    def _convert(field, feature, field_index, field_feature_index_dict):
-        if isinstance(feature, str):
-            field_feature_index = field_feature_index_dict[(field, feature)]
-            feature = 1
-        else:
-            field_feature_index = field_index
-
-        return "{}:{}:{}".format(field_index, field_feature_index, feature)
-
-    for col_index, col in enumerate(field_names):
-        df_new[col] = df_new[col].apply(lambda x: _convert(col, x, col_index+1, field_feature_dict))
-
-    # Move rating column to the first.
-    field_names.insert(0, col_rating)
-    df_new = df_new[field_names]
-
-    if filepath is not None:
-        np.savetxt(filepath, df_new.values, delimiter=' ', fmt='%s')
-
-    return df_new
-
+    def __init__(self, filepath=None):
+        if filepath is not None:
+            # Check the existence of path
+            import os
+            if not os.path.exists(filepath):
+                raise ValueError("The specified file path {} does not exist".format(self.filepath))
+        
+        self.filepath = filepath
+    
+    def fit(self, df, col_rating=DEFAULT_RATING_COL):
```

Comment:
This method is confusing. I would have thought that fit actually makes the transformation, but transform does. Also, the code `LibffmConveter().fit(df_feature, col_rating='rating')` looks a lot like Java. 
What if you have all the checkers in the init and then just user transform to compute the transformation? Would this be doable?

Reply:
This is to mimic the behaviour of `scikit-learn` preprocessing modules.

Reply:
I see, you are following [this example](https://github.com/scikit-learn/scikit-learn/blob/7b136e9/sklearn/preprocessing/data.py#L1812).

The thing is, sklearn are adding the fit method because they need it for pipelines. They say:
```
    def fit(self, X, y=None):
        """Do nothing and return the estimator unchanged
        This method is just there to implement the usual API and hence
        work in pipelines.
        Parameters
        ----------
        X : array-like
        """
        check_array(X, accept_sparse='csr')
        return self
```
I guess they are adding fit, just because many of their methods has a fit and because of that, they really need to have it so other elements like pipelines don't break. @msalvaris do you know if this is the case?

Reply:
Decided to remove `fit_transform` method as what you suggested. After a while (I mean after having a good sleep and breakfast 😄) I think this makes sense - indeed chaining the method is not very necessary in this class. I will still keep `fit` and `transform` methods for the reason that a fitted converter can be applied to a new dataframe with the same schema.

Reply:
makes sense